src/tests/Microsoft.PowerFx.Interpreter.Tests/RecalcEngineTests.cs

```diff
@@ -144,6 +144,26 @@ public void Recalc2()
             // Batched up (we don't double fire)            
             AssertUpdate("B-->20;C-->25;D-->22;");
         }
+        
+        [Fact]
+        public void DeleteFormula()
+        {
+            var engine = new RecalcEngine();
+
+            engine.UpdateVariable("A", 1);
+            engine.SetFormula("B", "A*10", OnUpdate);
+            engine.SetFormula("C", "B+5", OnUpdate);
+            engine.SetFormula("D", "B+A", OnUpdate);
+
+            Assert.Throws<InvalidOperationException>(() =>
+                engine.DeleteFormula("X"));
+
+            Assert.Throws<InvalidOperationException>(() =>
+                engine.DeleteFormula("B"));
+
+            engine.DeleteFormula("D");
+            Assert.False(engine.Formulas.TryGetValue("D", out var ret));
+        }

```

Comment:
Some other test cases:
- B fails, but if I delete C,D , can I then successfully delete B. 

- Can I readd B (perahps w/ different type) after deleting?
Ie, ensure that B really is fully deleted.